src/IIIFPresentation/Core/Response/HttpResponseMessageX.cs

```diff
@@ -46,4 +56,39 @@ public static bool IsJsonResponse(this HttpResponseMessage response)
         var mediaType = response.Content.Headers.ContentType?.MediaType;
         return mediaType != null && mediaType.Contains("json");
     }
+    
+    public static async Task<T?> ReadAsPresentationResponseAsync<T>(this HttpResponseMessage response,
```

Comment:
Behaviour seems unexpected from the name of this method when `ReadAsPresentationResponseAsync<Error>` as it will always throw.

I'd expect `ReadAsPresentationResponseAsync<Error>` to return an `Error` object if it finds it, rather than only doing that for certain status codes and throwing for others